### PR TITLE
Add configuration of username from metaflow_config.py

### PR DESCRIPTION
**Pull request by tfurmston**
This PR does the following:

* Allows for the username, as determined from the function `get_username`, to be configured through standard configuration functionality in `metaflow_config.py`. In particular, it allows for the username to be configured through Metaflow configuration files.

### CODE REVIEW

File: test/unit/test_util.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import os
```
Comment (savingoyal) [inline test/unit/test_util.py:1]: @tfurmston I don't think that these unit tests are executed by the CI harness. We already have tests that check if the user name is getting correctly passed around - https://github.com/Netflix/metaflow/blob/master/test/core/contexts.json#L7 and https://github.com/Netflix/metaflow/blob/master/test/core/metaflow_test/metadata_check.py#L32

**Review by savingoyal — CHANGES_REQUESTED** [review]
quick comments. otherwise lgtm!

File: test/unit/test_util.py (hunk 1 above)
Comment (tfurmston) [reply test/unit/test_util.py:1]: Ah, ok. Good to know. I did wonder about this point, but there were some other tests in the `unit` test folder that were structured in the same way. 

I should remove these new tests completely then?

Comment (savingoyal) [reply test/unit/test_util.py:1]: Yes - that will be super!

Comment (tfurmston) [reply test/unit/test_util.py]: cool. All done.

**Review by savingoyal — APPROVED** [review]

**Pull request merged**